src/Swift.Bindings/src/Demangler/StringSlice.cs

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Text;
+using System.Collections.Generic;
+
+namespace BindingsGeneration.Demangling {
+	/// <Summary>
+	/// Represents a a slice which includes facilities similar to StringStream, but
+	/// simplified and more performant. This is build to work specifically with the
+	/// demangler.
```

Comment:
I'm not sure if it's worth having a separate `StringSlice` class instead of using for example `(ReadOnly)Span` to represent a string slice. Span is heavily optimized inside .NET 9. Do you know what benefits this class has in the bigger picture of the tooling?

Reply:
StringSlice was written and in place before Span was ready. I'll look at replacing it.